dspace-spring-rest/src/test/java/org/dspace/app/rest/UriListParsingIT.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+
+import org.dspace.app.rest.builder.CollectionBuilder;
+import org.dspace.app.rest.builder.CommunityBuilder;
+import org.dspace.app.rest.builder.ItemBuilder;
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.DSpaceObject;
+import org.dspace.content.Item;
+import org.dspace.core.Constants;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/**
+ * This test class will test the UriList parsing method in the Utils class
+ */
+public class UriListParsingIT extends AbstractControllerIntegrationTest {
```

Comment:
this is an unit test not an integration test

Reply:
@abollini :  I've been looking at this a bit, and while this is on the edge between an IT and a Unit Test, I believe it's actually an IT within the context of our "spring-rest" webapp *as it requires data be created* (in the database), then mocks a request and verifies the results of the parsed results of that mocked request (based on the created data).

Unfortunately, as we currently stand in the "spring-rest" webapp, all our existing Unit Tests do *not* create content to test against.  Doing so would require creating a Unit Test framework that spins up a Kernel & database connection.

So, for the purposes of this PR, I'd recommend we either consider this an IT or schedule this for future work (via a JIRA ticket).  Does that seem reasonable?

Reply:
I will not comment on the short term, because time is short and the cure is long.

`<rant>`For the long term, *unit tests should not be touching the database*.  We have (in various projects) a lot of "unit" tests which spend more time testing the DBMS than the unit under test.  That is one reason they take so **long** to run.  We don't need another unit test framework; we need to destroy those we have and replace them with a good set of mocks and stubs.`</rant>`

Reply:
@mwoodiupui : I agree with your well meaning `<rant>`.  I was similarly pointing out that this test (as implemented) is more of an IT than a Unit Test.  If we wanted this to be a Unit Test, it would require rewriting / changing its purpose.  As-is, this `UriListParsingIT` requires database access for the `utils.constructDSpaceObjectList()` call to work properly (and that's the key method being tested here).  The only other way to achieve this test would be to build in more extensive mocks/stubs (which we don't yet have for the "spring-rest" project, and haven't required of other PRs yet).